java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -38,7 +41,7 @@
 import org.sonar.plugins.java.api.tree.MethodTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
-public class DefaultJavaFileScannerContext implements JavaFileScannerContext {
+public class DefaultJavaFileScannerContext implements JavaFileScannerContext, AnalyzerMessageReporter {
```

Comment:
I don't think we should use a dedicated interface, let's downcast in check implementation for this one case for now, we'll clean it up if required.